.ci/linux-steps.yaml

```diff
@@ -35,35 +36,6 @@ steps:
       sudo tar -C /opt/ -xvpf julia-1.10.4-linux-x86_64.tar.gz
     fi
 
-    # Install armadillo.
```

Comment:
The oldest supported version of Armadillo (10.8.2) was specifically used to make sure that contributed code would still compile in that situation.  I wouldn't object to a different way of testing that, but it was at least convenient because in the past there have been PRs that unknowingly break old Armadillo support.

And for ensmallen, we used the latest version just so whatever support is available there is quickly available here; so e.g. if we fix a bug there, as soon as it's merged and a release happens, we have that here.  So it can help accelerate broken PRs here, vs. waiting on the Debian package to update.  But, that doesn't happen often... so I don't mind removing it.

(Using apt for cereal and STB is definitely an improvement, those versions are basically arbitrary.)

Reply:
Ahhhhh.   Good context regarding Armadillo.  I will revert that.

For ensmallen I did this precisely I did *not* see you do what I have come to understand is common:  pick a tagged version from a GitHub release.  But now that you say it ... I know exactly where you are coming from.  You prefer _the latest version, automagically_ without having updated tags.  I think I once knew how to get that from a GitHub Release but I may need to ponder where I used that / learned. 

So will revert both those.